go/vt/vtgate/planbuilder/testdata/dml_cases.json

```diff
@@ -5819,6 +5819,181 @@
+    "query": "update user as u, user_extra as ue set u.col = ue.col where u.id = ue.id",
+    "plan": {
+      "QueryType": "UPDATE",
+      "Original": "update user as u, user_extra as ue set u.col = ue.col where u.id = ue.id",
+      "Instructions": {
+        "OperatorType": "DMLWithInput",
+        "TargetTabletType": "PRIMARY",
+        "BindVars": [
+          "0:map[ue_col:1]"
+        ],
+        "Offset": [
+          "0:[0]"
+        ],
+        "Inputs": [
+          {
+            "OperatorType": "Join",
+            "Variant": "Join",
+            "JoinColumnIndexes": "R:0,L:0",
+            "JoinVars": {
+              "ue_id": 1
+            },
+            "TableName": "user_extra_`user`",
+            "Inputs": [
+              {
+                "OperatorType": "Route",
+                "Variant": "Scatter",
+                "Keyspace": {
+                  "Name": "user",
+                  "Sharded": true
+                },
+                "FieldQuery": "select ue.col, ue.id from user_extra as ue where 1 != 1",
+                "Query": "select ue.col, ue.id from user_extra as ue for update",
+                "Table": "user_extra"
+              },
+              {
+                "OperatorType": "Route",
+                "Variant": "EqualUnique",
+                "Keyspace": {
+                  "Name": "user",
+                  "Sharded": true
+                },
+                "FieldQuery": "select u.id from `user` as u where 1 != 1",
+                "Query": "select u.id from `user` as u where u.id = :ue_id for update",
+                "Table": "`user`",
+                "Values": [
+                  ":ue_id"
+                ],
+                "Vindex": "user_index"
+              }
+            ]
+          },
+          {
+            "OperatorType": "Update",
+            "Variant": "IN",
+            "Keyspace": {
+              "Name": "user",
+              "Sharded": true
+            },
+            "TargetTabletType": "PRIMARY",
+            "Query": "update `user` as u set u.col = :ue_col where u.id in ::dml_vals",
+            "Table": "user",
+            "Values": [
+              "::dml_vals"
+            ],
+            "Vindex": "user_index"
+          }
+        ]
+      },
+      "TablesUsed": [
+        "user.user",
+        "user.user_extra"
+      ]
+    }
+  },
+  {
+    "comment": "update with multi table join with multi target having dependent column update",
+    "query": "update user, user_extra ue set user.name = ue.id + 'foo', ue.bar = user.baz where user.id = ue.id and user.id = 1",
+    "plan": {
+      "QueryType": "UPDATE",
+      "Original": "update user, user_extra ue set user.name = ue.id + 'foo', ue.bar = user.baz where user.id = ue.id and user.id = 1",
+      "Instructions": {
+        "OperatorType": "DMLWithInput",
+        "TargetTabletType": "PRIMARY",
+        "BindVars": [
+          "0:map[ue_id:1]",
```

Comment:
nitpick: `map` is not needed here

Reply:
This is coming from printing out the map for each index.
Another format could be 

```
"0:map[ue_id:1]" => "0:[{ue_id 1}]",
"1:map[user_baz:3]" => "1:[{user_baz 3}]"
```